### PR TITLE
Test case for the public jira https://wso2.org/jira/browse/CAPIMGT-12

### PR DESCRIPTION
**Pull request by rmsamitha**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    @BeforeClass(alwaysRun = true)
+    public void init() throws APIManagerIntegrationTestException {
+        super.init();
+        try {
+            userManagementClient.addUser(TEST_USER_1, TEST_USER_1_PASSWORD, new String[] { "admin" }, "manager");
+            userManagementClient.addUser(TEST_USER_2, TEST_USER_2_PASSWORD, new String[] { "admin" }, "manager");
+        } catch (UserAdminUserAdminException e) {
+            log.error("Error while creating the test users");
+            //Fail the test case.
+            assertTrue(false, e.getMessage());
```
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java]: can you use Assert.fail(e.getMessage()) here

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        } catch (RemoteException e) {
+            log.error("Error while creating the test users");
+            //Fail the test case.
+            assertTrue(false, e.getMessage());
```
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java]: can you use Assert.fail(e.getMessage()) here

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                        + " back URL of second user's application.");
+    }
+
+    /**
+     * Add new application and Generate production keys for each the users (similar
+     * names for both the applications)
+     *
+     * @throws Exception
+     */
+    public void addAppsAndGEnerateKeys() throws Exception {
```
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java]: this should be renamed to addAppsAndGenerateKeys() right?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java (hunk 1 above)
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java]: Fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java (hunk 2 above)
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java]: Fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java (hunk 3 above)
Comment (rmsamitha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/CAPIMGT12CallBackURLOverwriteTestCase.java]: Fixed

**Pull request merged**